### DISCUSSION TITLE
[Solved] Add a button/option to build and run directly on plugged Android device

### DISCUSSION THREAD

**Discussion by hsandt** [godotengine/godot-proposals#8948] [Editor]
I was using Unity and Renpy and both have an option/button to Build, Install and Run directly on a plugged Android device, which I find convenient for testing: no need to copy the apk file to device after each build.

It would be nice to have a similar option in the Export for Android popup. Or even a top-right button like the "Remote Debug" for web.

For now I'm trying `adb install mygame.apk` in command line but it doesn't work, I may just end up copying the apk to device storage and installing it manually.

**Comment by Calinou** [comment]
This is already implemented as one-click deploy if your export preset is marked as Runnable: https://docs.godotengine.org/en/stable/tutorials/export/one-click_deploy.html

**Reply by hsandt** [reply]
Thanks! I found it in the meantime but somewhat forgot about this discussion and so I never updated it as solved. I don't see a way to mark a discussion as "solved" since it's apparently meant to be something open so I'll just add it to the title.

**Reply by Calinou** [reply]
>  I don't see a way to mark a discussion as "solved" since it's apparently meant to be something open so I'll just add it to the title.

GitHub Discussions can be configured to be in Q&A mode, but this one isn't.
